core/src/main/java/hudson/widgets/buildsearch/BuildSearchParamProcessorFactory.java

```diff
@@ -0,0 +1,109 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.widgets.buildsearch;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.widgets.buildsearch.processors.DateProcessorFactory;
+import hudson.widgets.buildsearch.processors.DescriptionProcessorFactory;
+import hudson.widgets.buildsearch.processors.NameProcessorFactory;
+import hudson.widgets.buildsearch.processors.ResultProcessorFactory;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Implementations of this class create {@link BuildSearchParamProcessor} instances from a set
+ * of {@link BuildSearchParams}. The set of {@link BuildSearchParamProcessor} instances are then used
+ * by the {@link hudson.widgets.HistoryPageFilter} class to filter the build history (via an instance of
+ * {@link BuildSearchParamProcessorList}).
+ *
+ * <p>
+ * Each {@link BuildSearchParamProcessor} implementation processes one or more search terms e.g. the
+ * {@link BuildSearchParamProcessor} instance created by the {@link NameProcessorFactory} checks the
+ * build {@link hudson.model.Queue.Item} or {@link hudson.model.Run} name using the "name:" search token(s)
+ * provided in the search, while the {@link DateProcessorFactory} checks date using one or both of the
+ * "date-from:" and "date-to:" tokens.
+ *
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+public abstract class BuildSearchParamProcessorFactory implements ExtensionPoint {
+
+    private static List<BuildSearchParamProcessorFactory> coreParamProcessorFactories = new ArrayList<BuildSearchParamProcessorFactory>();
+
+    static {
+        // Statically adding the core BuildSearchParamProcessorFactory impls Vs adding via @Extension.
+        // Makes testing easier coz no need to create a JenkinsRule instance + put in test harness etc.
```

Comment:
> You do not need JenkinsRule to call ExtensionList.lookup: from a unit test it will look for implementations in the classpath.

Fairly sure I tried this and got an empty list. Will try again and see.